### PR TITLE
Adding Invoke-Robocopy to internal commands

### PR DESCRIPTION
**Pull request by potatoqualitee**
I mostly wanted to restart the conversation around `Move-DbaDbFile`

Let's just use robocopy, and not try 43 different ways to Sunday for v1

This can be simple and we can pile on later. @ClaudioESSilva had some old code that we can explore, perhaps.

We use `Invoke-Command2 -Raw` and the definition of `Invoke-Robocopy` within the scriptblock

http://www.smattie.com/2016/05/02/how-to-use-robocopy-to-move-sql-files/ which shows active output (unsure if it'll work with the way we call it)

refs
`Robocopy e:\mssql\backup\bkAP.bak \\DestinationServer\f$\mssql\backup\bkAP.bak /MIR /SEC /Log:c:\temp\bkAP.log`

so `/MIR /SEC` by default? Retry and wait that are settable `/R:1 /W:5`

http://www.sqlservercentral.com/articles/Disaster+Recovery+%28DR%29/75860/

### CODE REVIEW

**Comment by niphlod** [comment]
Use copy-item, not robocopy. Please.

Il gio 27 set 2018, 16:49 Chrissy LeMaire <notifications@github.com> ha
scritto:

**Comment by potatoqualitee** [comment]
copy-item blows, why do you like it

**Comment by potatoqualitee** [comment]
I love robocopy because it retries and it just works. I moved some files around tonight by using `robocopy source destination /MIR /Z /SEC /XE "System Volume Information"` and it worked so well. I even got interrupted by some file perm issues (hence System Volume Information) and I resumed just like that. The whole experience was so fast.

**Comment by niphlod** [comment]
I guess we'll agree on disagreeing

Il gio 27 set 2018, 20:39 Chrissy LeMaire <notifications@github.com> ha
scritto:

**Comment by gwojan** [comment]
Robocopy is a great tool but it's Windows only. What happens when serious thought is given to porting dbatools to non-Windows platforms?

**Comment by potatoqualitee** [comment]
we can add it at that time, or detect by os and use rsync or copy-item. rsync has that resiliency i like about robocopy, tho i dont recall if it's zippy.

**Comment by potatoqualitee** [comment]
convincing argument that bums me out ;)

**Comment by niphlod** [comment]
1) collect requirements 2) choose the best tool for the job

File: internal/functions/Invoke-Robocopy.ps1 [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                    }
+                    else {
+                        $Output = & $FilePath $ArgumentList 2>&1
+                    }
+                }
+                catch {
+                    throw $_
+                }
+
+                Write-PSFMessage -Level Verbose -Message 'Finished, processing output'
```
[the remark this answers is not in the document]
Comment (potatoqualitee) [reply internal/functions/Invoke-Robocopy.ps1]: no, i took it from my other module, ty. ill close it anyway :(

**Pull request closed without merging**